### DISCUSSION TITLE
Is it possible to plot a continuous stream of data

### DISCUSSION THREAD

**Discussion by hmf** [HanSolo/charts#68] [Q&A] [answered]
I would like to know is it is possible to plot time-series data continuously to an `XYChart`. Any example I can use as a starting point? My apologies if this is obvious but my JavaFX is very shaky.  

TIA.

**Comment by HanSolo** [comment, marked as answer]
Yep that‘s possible but you need to present the data in the right way. I would use a series of items and instead of always adding new items I would shift the values in the list, drop the oldest one and add the latest one at the front.

**Reply by hmf** [reply]
If I understand you correctly I need to take the `ArrayList<TYChartItem>` of length `n`, shift the last `m`  values to the the first `m`values and then fill in the rest of the `n-m` slots with the new items. But don't I have to refresh the plot?

**Comment by HanSolo** [comment]
Correct, the chart should automatically listen to changes in items
